test/testSimplifier.ts

```diff
@@ -0,0 +1,153 @@
+import * as assert from 'assert';
+import { ModeName } from '../src/mode/mode'
+import { Position } from '../src/motion/position';
+import { ModeHandler } from '../src/mode/modeHandler';
+import { TextEditor } from '../src/textEditor';
+import { assertEqualLines } from './testUtils';
+
+
+interface TestObject {
+    start: string[],
+    keysPressed: string,
+    end: string[],
+    endMode?: ModeName
+}
+
+class TestObjectHelper {
+    startPosition = new Position(0, 0);
+    endPosition = new Position(0, 0);
+    private _isValid = false;
+    
+    constructor(private _testObject: TestObject) {
+        this._parse(_testObject);
+    }
+
+    public get isValid(): boolean {
+        return this._isValid;
+    }
+    
+    private _setStartCursorPosition(lines: string[]) {
+        let result = this._getCursorPosition(lines);
+        this.startPosition = result.position;
+        return result.success;
+    }
+
+    private _setEndCursorPosition(lines: string[]) {
+        let result = this._getCursorPosition(lines);
+        this.endPosition = result.position;
+        return result.success;
+    }
+    
+    private _getCursorPosition(lines: string[]) {
+        let ret = { success: false, position: new Position(0, 0) };
+        lines.forEach((line, lineIdx) => {
+            let columnIdx = line.indexOf('|');
+            if (columnIdx >= 0) {
+                ret.position = ret.position.setLocation(lineIdx, columnIdx);
+                ret.success = true;
+            }
+        });
+        return ret;
+    }
+
+    private _parse(t: TestObject) {
+        if (!this._setStartCursorPosition(t.start)) {
+            this._isValid = false;
+            return;
+        }
+        if (!this._setEndCursorPosition(t.end)) {
+            this._isValid = false;
+            return;
+        }
+
+        this._isValid = true;
+    }
+
+    public asVimInputText(): string[] {
+        let ret = 'i' + this._testObject.start.join('\n').replace('|', '');
+        return ret.split('');
+    }
+
+    public asVimOutputText(): string[] {
+        let ret = this._testObject.end.slice(0);
+        ret[this.endPosition.line] = ret[this.endPosition.line].replace('|', '');
+        return ret;
+    }
+
+    /** Returns a sequence of Vim movement characters 'hjkl' as a string array
+     * which will move the cursor to the start position given in the test. */
+    public getKeyPressesToMoveToStartPosition(): string[] {
+        let ret: string = '';
+        let h: number, j: number, k: number, l: number;
```

Comment:
You can initialize multiple variables all in one go:

`let h = 0, j = 0, k = 0, l = 0;` 

Also if you initialize a variable when you create it, you don't need to declare a type for it, because it will be inferred by TypeScript.